zerocopy-derive/tests/ui/padding_check.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2019 The Fuchsia Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#[macro_use]
+extern crate zerocopy;
+
+fn main() {}
+
+// These tests exercise our checks for padding bytes in types.
+
+#[repr(C, align(4))]
+struct AlignedU32(u32);
+
+#[derive(AsBytes)]
+#[repr(C)]
+struct AsBytesStruct {
+    foo: u8,
+    bar: AlignedU32,
```

Comment:
Just so I understand correctly, are you using this type as opposed to `u32` because the latter's alignment isn't 4 on all platforms?

Reply:
Yep, the alignment of `u32` could be as low as 1 so I manually set the align to 4 to guarantee padding bytes.